### PR TITLE
Allow specifying specific tag when downloading an image

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

**Comment by KKoukiou** [comment]
search-image-tag
error-download-tag

**Comment by KKoukiou** [comment]
The error message as all libpod error is  not good. But this is a matter of the API and not this PR.

File: src/Images.jsx [hunk 1]
```diff
@@ -73,7 +77,7 @@ class Images extends React.Component {
                     let error = (
                         <React.Fragment>
                             <strong>
-                                {cockpit.format(_("Failed to download image $0"), imageName)}
+                                {cockpit.format(_("Failed to download image $0:$1"), imageName, imageTag || "latest")}
```
Comment (Gundersanne) [inline src/Images.jsx]: Is no tag equivalent to `:latest` ?

**Comment by Gundersanne** [comment]
> search-image-tag

I don't mind the way it looks but would it make more sense to have the tag input box next to the image name on the right side?

File: test/check-application [hunk 2]
```diff
@@ -169,51 +169,119 @@ class TestApplication(testlib.MachineCase):
...
+        prepare()
+
+        self.login_and_go("/podman")
+        b.wait_present("#app")
 
-        # Search for an non-existing image
-        b.wait_present("#search-image-dialog-name")
-        b.set_input_text("#search-image-dialog-name", "foobar")
-        b.wait_present("div.modal-body:contains(No results for foobar)")
+        dialog = DownloadImageDialog('my-busybox')
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply test/check-application]: > > search-image-tag
> 
> I don't mind the way it looks but would it make more sense to have the tag input box next to the image name on the right side?

I don't mind either. I blindly copied design from docker dialog TBH.

File: src/Images.jsx (hunk 1 above)
Comment (KKoukiou) [reply src/Images.jsx]: Right, if you don't specify tag podman will pull latest tag by default.
From `man podman pull`:
`If an image tag is not specified, podman pull defaults to the image with the latest tag (if it exists) and pulls it.`

Comment (Gundersanne) [reply src/Images.jsx]: cool! I'm good with this then :)

**Comment by KKoukiou** [comment]
> Can you add the Closes #xxxx thing?

Done

**Comment by KKoukiou** [comment]
I just changed the commit message, and now it failed with missing method!?!? https://fedorapeople.org/groups/cockpit/logs/pull-65-20190311-133535-c889e166-cockpit-fedora-29/TestApplication-testDownloadImage-fedora-29-127.0.0.2-2201-FAIL.png
I wish re-trigger to see if it persists, but this should not happen :/

**Comment by KKoukiou** [comment]
The API had changed and tests are affected by it. I will repost with adjustments.

**Pull request merged**